services/cloudapi/cloudapinic/nic.go

```diff
@@ -0,0 +1,270 @@
+package cloudapinic
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+)
+
+type Service struct {
+	Client *ionoscloud.APIClient
+	Meta   interface{}
+	D      *schema.ResourceData
+}
+
+func (fs *Service) Get(ctx context.Context, datacenterID, serverID string, depth int32) ([]ionoscloud.Nic, error) {
+	nics, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsGet(ctx, datacenterID, serverID).Depth(depth).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return nil, err
+	}
+	if apiResponse.HttpNotFound() || nics.Items == nil || len(*nics.Items) == 0 {
+		log.Printf("[DEBUG] no nics found for datacenter %s, server %s", datacenterID, serverID)
+		return nil, nil
+	}
+	return *nics.Items, nil
+}
+
+func (fs *Service) FindById(ctx context.Context, datacenterId, serverId, ID string, depth int32) (*ionoscloud.Nic, *ionoscloud.APIResponse, error) {
+	nic, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsFindById(ctx, datacenterId, serverId, ID).Depth(depth).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			log.Printf("[DEBUG] no nic found for datacenter %s, server %s", datacenterId, serverId)
+		}
+		return nil, apiResponse, err
+	}
+	return &nic, apiResponse, nil
+}
+
+func (fs *Service) Delete(ctx context.Context, datacenterID, serverID, ID string) (*ionoscloud.APIResponse, error) {
+	apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsDelete(ctx, datacenterID, serverID, ID).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return apiResponse, err
+	}
+	// Wait, catching any errors
+	_, errState := cloudapi.GetStateChangeConf(fs.Meta, fs.D, apiResponse.Header.Get("Location"), schema.TimeoutDelete).WaitForStateContext(ctx)
+	if errState != nil {
+		return apiResponse, fmt.Errorf("an error occured while waiting for state change dcId: %s, server_id: %s, ID: %s, Response: (%w)", datacenterID, serverID, ID, errState)
```

Comment:
@cristiGuranIonos please include the `nic` keyword somewhere in this error messages because it won't be clear in all those log messages that we are talking about a `nic` resource.